modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/resource/OpenAPIResourceImpl.java

```diff
@@ -119,7 +119,37 @@ public void setConfiguration(
 
 		OpenAPI openAPI = openApiContext.read();
 
+		OpenAPISchemaFilter schemaFilter = null;
```

Comment:
I find the code difficult to follow with so many related conditions. Can we reorganize it to improve readability?

I see some options but I don't see a great one:
- Extract a _mergeOpenAPISchemaFilter method that handles the creation of the final OpenAPISchemaFilter to use and get rid of some conditions.
- Initialize the schemaFilter to the openAPISchemaFilter and get rid of the else and maybe some conditions.
- Reorganize the code so we can have some early returns with an error or basic response like when dtoClassNames is empty.
- Use some sort of Null Object pattern, maybe with OpenAPISchemaFilter.

I think a lot of these variables that we check for null are related and we can improve the code by somehow being aware of these relationships.

Reply:
I have updated the commit with some of the suggestions. Please, have a look at it.

Any suggestion is welcomed again!

Reply:
I like it much more like this. I think it has improved readability

Reply:
Great. You are right, it has improved a lot